### PR TITLE
Add EightBit encoding

### PR DESCRIPTION
**Pull request by etrepum**

### CODE REVIEW

**Comment by kizzx2** [comment]
Thanks for the PR! :+1:

**Comment by etrepum** [comment]
Let me know if there's anything else I can provide to help you with this. I was going through some old code that encodes URLs as QR codes using haskell-qrencode which has a C dependency so I figured I'd see if I can make it work in a way that's much easier to compile on platforms like Windows.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /dist/
 /.stack-work/
+/hello.pgm
```
[the remark this answers is not in the document]
Comment (etrepum) [reply .gitignore]: This file is produced if you run the example script, it seemed appropriate

File: Codec/Binary/QRCode.hs [hunk 2]
```diff
@@ -54,7 +55,8 @@ encode ver ecl mode input =
         inputEncode = case mode of
             Numeric -> N.encode
             Alphanumeric -> A.encode
-            _ -> undefined
+            EightBit -> E.encode
+            Kanji -> undefined
```
[the remark this answers is not in the document]
Comment (etrepum) [reply Codec/Binary/QRCode.hs]: It seemed better to make this a complete pattern match even if the function still isn’t total. If the type changes to add new encoding modes, this style will be helpful.

[the remark this answers is not in the document]
Comment (kizzx2) [reply Codec/Binary/QRCode.hs]: Fair enough

**Pull request merged**